File: test-fixtures/codebase/Sample.kt
package sample

/**
 * Sample Kotlin file for testing.
 * Demonstrates various Kotlin features including annotations.
 */

/**
 * A greeter class that holds a name.
 * Used for generating greeting messages.
 */
data class Greeter(private val name: String) {
    /**
     * Returns a greeting message.
     */
    fun greet(): String {
        return "Hello, $name!"
    }

    private fun privateMethod(): String {
        return name
    }
}

/**
 * Internal helper class.
 */
internal class PrivateHelper(private val value: Int) {
    fun calculate(x: Int): Int {
        return value + x
    }

    private fun internalHelper(): Int {
        return value * 2
    }
}

/**
 * Add two numbers together.
 * @param a First number
 * @param b Second number
 * @return The sum of a and b
 */
@Suppress("unused")
fun add(a: Int, b: Int): Int {
    return a + b
}

private fun privateFunction(): Int {
    return 42
}

/**
 * Process data with validation.
 */
@Deprecated("Use processDataV2 instead", ReplaceWith("processDataV2(data)"))
fun processData(data: String): String? {
    return if (data.isEmpty()) null else data.uppercase()
}

/**
 * Public interface for services.
 */
interface Service {
    /**
     * Executes the service.
     */
    fun execute()
}

/**
 * Object declaration for utilities.
 */
@JvmStatic
object MathUtils {
    /**
     * Multiplies two integers.
     */
    fun multiply(a: Int, b: Int): Int {
        return a * b
    }
}

File: test-fixtures/codebase/sample.php
<?php

declare(strict_types=1);

namespace Sample;

/**
 * Sample PHP file for testing.
 * Demonstrates various PHP 8 features including attributes.
 */

/**
 * A greeter class that holds a name.
 * Used for generating greeting messages.
 */
#[AllowDynamicProperties]
class Greeter
{
    private string $name;

    /**
     * Creates a new Greeter instance.
     */
    public function __construct(string $name)
    {
        $this->name = $name;
    }

    /**
     * Returns a greeting message.
     */
    public function greet(): string
    {
        return "Hello, {$this->name}!";
    }

    private function privateMethod(): void
    {
        // Private method
    }
}

/**
 * Internal helper class.
 */
class PrivateHelper
{
    private int $value;

    public function __construct(int $value)
    {
        $this->value = $value;
    }

    public function calculate(int $x): int
    {
        return $this->value + $x;
    }

    private function internalHelper(): int
    {
        return $this->value * 2;
    }
}

/**
 * Add two numbers together.
 *
 * @param int $a First number
 * @param int $b Second number
 * @return int The sum
 */
#[Pure]
function add(int $a, int $b): int
{
    return $a + $b;
}

/**
 * Process data with validation.
 */
#[Deprecated('Use processDataV2 instead')]
function processData(string $data): ?string
{
    if (empty($data)) {
        return null;
    }
    return strtoupper($data);
}

/**
 * Public interface for services.
 */
interface Service
{
    /**
     * Executes the service.
     */
    public function execute(): void;
}

/**
 * Trait for data processing.
 */
#[Attribute]
trait DataProcessorTrait
{
    /**
     * Processes the given input.
     */
    public function process(string $input): void
    {
        echo $input;
    }
}

/**
 * Enum for status values.
 */
enum Status: string
{
    case Active = 'active';
    case Inactive = 'inactive';
}
